src/util/command_line_handling.cpp

```diff
@@ -667,8 +667,7 @@ namespace hpx { namespace util
             }
         }
 
-        if ((vm.count("hpx:connect") || mode_ == hpx::runtime_mode_connect) &&
-            hpx_host == "127.0.0.1")
+        if (hpx_host == "127.0.0.1" || hpx_host == "localhost")
```

Comment:
I'd better check that. we have some issues with the hostnames that this resolves, but it may introduce new problems. srun and manual start behave differently and for one I needed IB ip addresses, for the other I needed TCP ip addresses. It's a bit of a mess and I'm not sure what to do for the general case. I'll test more and see if hermione/rostam can be fixed without breaking my local usage.